kolibri/plugins/coach/assets/src/views/lessons/LessonResourceSelectionPage/ContentCardList.vue

```diff
@@ -90,19 +74,85 @@
 
 <script>
 
+  import { computed, ref } from 'vue';
+  import urls from 'kolibri/urls';
+  import client from 'kolibri/client';
+  import useUser from 'kolibri/composables/useUser';
+  import BookmarksResource from 'kolibri-common/apiResources/BookmarksResource';
   import commonCoreStrings from 'kolibri/uiText/commonCoreStrings';
+  import AccessibleFolderCard from 'kolibri-common/components/Cards/AccessibleFolderCard';
+  import AccessibleResourceCard from 'kolibri-common/components/Cards/AccessibleResourceCard';
   import { ViewMoreButtonStates } from '../../../constants/index';
-  import LessonContentCard from './LessonContentCard';
 
   export default {
     name: 'ContentCardList',
     components: {
-      LessonContentCard,
+      AccessibleResourceCard,
+      AccessibleFolderCard,
     },
     mixins: [commonCoreStrings],
     setup() {
+      // Map of contentnode_id to bookmark resource ID
+      const bookmarks = ref({});
+      // Map of contentNode IDs to bookmark resource IDs
+      const bookmarkedContentNodeIds = computed(() => Object.keys(bookmarks.value));
+      const { currentUserId } = useUser();
+
+      /**
+       * Fetch bookmarks and store them in the bookmarks ref mapping
+       * their contentnode_id to the bokomark's own ID.
+       * The contentnode_id is used for creating it, but we need the
+       * bookmark's ID to delete it.
+       */
+      function getBookmarks() {
+        BookmarksResource.fetchCollection({ force: true }).then(data => {
+          bookmarks.value = data.reduce((memo, bookmark) => {
+            memo[bookmark.contentnode_id] = bookmark.id;
+            return memo;
+          }, {});
+        });
+      }
+
+      function deleteBookmark(contentnode_id) {
+        client({
+          method: 'delete',
+          url: urls['kolibri:core:bookmarks_detail'](contentnode_id),
+        }).then(() => {
+          getBookmarks();
+        });
+      }
+
+      function addBookmark(contentnode_id) {
+        client({
+          method: 'post',
+          url: urls['kolibri:core:bookmarks_list'](),
+          data: {
+            contentnode_id: contentnode_id,
+            user: currentUserId.value,
+          },
+        }).then(() => {
+          getBookmarks();
+        });
+      }
+
+      function isBookmarked(contentnode_id) {
```

Comment:
Just wonder if it would be better to check the existence of `content.bookmark` object to verify if the resource has been bookmarked? For example we use this to show the "bookmarked x days ago" in the SelectFromBookmarks subpage [here](https://github.com/nucleogenesis/kolibri/blob/01fd4480f1ef4f9a54ac1d07da2b2522c6bcb4df/kolibri/plugins/coach/assets/src/views/common/resourceSelection/subPages/SelectFromBookmarks.vue#L83).

With this we avoid needing to load all bookmarks each time this component is mounted. Because this component will be remounted each time we navigate through the folder tree/bookmarks/search results and I think we can save us from making this fetch all these times.

Reply:
Thanks @AlexVelezLl - I'll ping you as I work on it in more detail addressing [this issue](https://github.com/learningequality/kolibri/issues/13077) in another PR